src/langcheck/metrics/model_manager/_model_management.py

```diff
@@ -0,0 +1,257 @@
+                    )
+                loader_func = model_setting.pop('loader_func', None)
+                if loader_func not in VALID_LOADER_FUNCTION:
+                    raise ValueError(
+                        f'loader type should in {VALID_LOADER_FUNCTION}')
+
+                # Check that the model and revision are available on the Hugging
+                # Face Hub
+                model_name = model_setting.pop('model_name')
+                revision = model_setting.pop('revision', None)
+                if not check_model_availability(model_name, revision):
+                    raise ValueError(
+                        f'Cannot find {model_name} with {revision} and Huggingface Hub'  # NOQA:E501
+                    )
+
+    def __set_model_for_metric(self, language: str, metric: str,
+                               model_name: str, loader_func: str,
+                               **kwargs) -> None:
+        '''
+        Set model for specified metric in specified language.
+
+        Args:
+            language: The name of the language
+            metric: The name of the evaluation metric
+            model_name: The name of the model
+            loader_func: The loader function of the model
+            tokenizer_name: (Optional) The name of the tokenizer
+            revision: (Optional) A version string of the model
+        '''
+        config_copy = deepcopy(self.config)
+        try:
+            if language not in VALID_LANGUAGE:
+                raise KeyError('Language {language} not supported yet')
+
+            if metric not in VALID_METRICS:
+                raise KeyError(
+                    f'Metric {metric} not supported for language {language} yet'
+                )
+
+            # Initialize the configuration for the language and metric if it
+            # doesn't exist
+            if self.config.get(language) is None:
+                self.config[language] = {}
+            if self.config.get(language).get(metric) is None:
+                self.config[language][metric] = {}
+
+            detail_config = self.config[language][metric]
+            # Set the loader function and model name
+            detail_config['loader_func'] = loader_func
+            detail_config['model_name'] = model_name
+
+            # If tokenizer_name is different from model_name
+            tokenizer_name = kwargs.pop('tokenizer_name', None)
+            if tokenizer_name:
+                detail_config['tokenizer_name'] = tokenizer_name
+            # If model's revision is pinned
+            revision = kwargs.pop('model_revision', None)
+            if revision:
+                detail_config['revision'] = revision
+
+            # Validate the change
+            if ModelManager.validate_config(self.config,
+                                            language=language,
+                                            metric=metric):
```

Comment:
This doesn't return anything (just raises an exception if something goes wrong) so I think we just need to call it?